### PR TITLE
dev to main

### PR DESCRIPTION
**Pull request by dhivahari-vivek-cdot**
not many code changes. just those try catches. exception handling. functionality remains the same. all changes are in the GCP_cloud_functions dir
edit: also made corrections in the README regarding using configmaps for config.py since I don't think we can do it outside kubernetes (at least for now, will continue looking into that)

### CODE REVIEW

File: GCP_cloud_functions/rsu-to-raw-ingest/rsu-to-raw-ingest.py [hunk 1]
```diff
@@ -14,17 +14,30 @@ def rsu_to_raw_ingest(event, context):
...
     client = storage.Client()
-    raw_bucket = client.get_bucket(config.config_vars['raw_ingest_id'])
-    raw_blob = raw_bucket.blob(str(datetime.datetime.now()))
-    raw_blob.upload_from_filename(filename='RSU-ND-clean.json')
 
-    # logging raw ingest upload
-    current_time = datetime.datetime.now()
-    log_message = Template('Raw ingest updated with new data at $time')
-    logging.info(log_message.safe_substitute(time=current_time))
+    try:
```
Comment (payneBrandon) [inline GCP_cloud_functions/rsu-to-raw-ingest/rsu-to-raw-ingest.py:19]: I'm not sure we need this outter try/catch. Will the datetime, Template, or logging methods here ever throw an error? 

I see the client.get_bucket throws an exception if not found so we definitely want to catch that, but these methods above seem to be safe.

File: GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py [hunk 2]
```diff
@@ -37,29 +37,37 @@ def is_json_clean(rsu_data):
     return isJSONclean
 
 def raw_to_data_lake(event, context):
-    """Triggered by new upload to the raw ingest storage bucket.
-    Function retrieves this new upload and checks for cleanliness
-    before sending clean data to the data lake bucket.
+    """Triggered from a message on a Cloud Pub/Sub topic.
```
Comment (payneBrandon) [inline GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py:40]: Let's keep a description in here of what the function does.

File: GCP_cloud_functions/rsu-to-raw-ingest/rsu-to-raw-ingest.py (hunk 1 above)
Comment (dhivahari-vivek-cdot) [reply GCP_cloud_functions/rsu-to-raw-ingest/rsu-to-raw-ingest.py:19]: From my understanding, the outer try-catch is if the Cloud Function fails to deploy for some reason, while the inner try-catch catches the bucket stuff. The official docs I've seen use this approach in their code examples.

File: GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py [hunk 3]
```diff
@@ -37,29 +37,37 @@ def is_json_clean(rsu_data):
...
-    lake_bucket = client.get_bucket(config.config_vars['data_lake_id'])
-
-    # retrieving the latest blob upload to the bucket
-    blob = raw_bucket.get_blob(event['name'])
-    data_string = blob.download_as_bytes()
-    json_data = ndjson.loads(data_string)
-    if is_json_clean(json_data) is True:
-        raw_bucket.copy_blob(blob, lake_bucket)
+    
+    try:
```
Comment (payneBrandon) [inline GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py:48]: As before, I don't think we need this outter try/catch

File: GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py [hunk 4]
```diff
@@ -37,29 +37,37 @@ def is_json_clean(rsu_data):
...
+        # logging cloud function trigger
         current_time = datetime.datetime.now()
-        log_message = Template('Data lake updated at $time')
+        log_message = Template('Cloud Function "raw-to-data-lake" was triggered at $time')
         logging.info(log_message.safe_substitute(time=current_time))
+        
+        try:
+            raw_bucket = client.get_bucket(config.config_vars['raw_ingest_id'])           
+            lake_bucket = client.get_bucket(config.config_vars['data_lake_id'])
+            blob = client.get_bucket(event['bucket']).get_blob(event['name'])
```
Comment (payneBrandon) [inline GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py:57]: Shouldn't we just reuse the `raw_bucket` object we pulled from above?

File: GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py [hunk 5]
```diff
@@ -37,29 +37,37 @@ def is_json_clean(rsu_data):
...
+            data_string = blob.download_as_string()
```
Comment (payneBrandon) [inline GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py:58]: I thought we fixed this in the last PR? The `download_as_string` method is deprecated.

File: GCP_cloud_functions/lake-to-data-warehouse/lake-to-data-warehouse.py [hunk 6]
```diff
@@ -6,9 +6,7 @@
 import config
 
 def rsu_data_warehouse_bucket(event, context):
-    """Triggered by new upload to the data lake storage bucket. 
-    Function retrieves this new upload and publishes it to the 
-    short-term warehouse Pub/Sub as a byte string.
+    """Triggered from a message on a Cloud Pub/Sub topic.
```
Comment (payneBrandon) [inline GCP_cloud_functions/lake-to-data-warehouse/lake-to-data-warehouse.py:9]: I think somewhere we're losing some of the new code you added last PR. We need to keep including this description information and not fall back to the previous description.

File: GCP_cloud_functions/lake-to-data-warehouse/lake-to-data-warehouse.py [hunk 7]
```diff
@@ -17,13 +15,30 @@ def rsu_data_warehouse_bucket(event, context):
...
-    topic = publisher.topic_path(config.config_vars['project_id'], config.config_vars['data_warehouse_id'])
-    blob = client.get_bucket(event['bucket']).get_blob(event['name'])
-    data_string = blob.download_as_string()
-    future = publisher.publish(topic,data_string)
-    print(future.result())
+    try:
+        # logging cloud function trigger
+        current_time = datetime.datetime.now()
+        log_message = Template('Cloud Function "lake-to-data-warehouse" was triggered at $time')
+        logging.info(log_message.safe_substitute(time=current_time))
```
Comment (payneBrandon) [inline GCP_cloud_functions/lake-to-data-warehouse/lake-to-data-warehouse.py:22]: Same as before, this outer try/catch doesn't seem to be needed

File: GCP_cloud_functions/lake-to-data-warehouse/lake-to-data-warehouse.py (hunk 6 above)
Comment (dhivahari-vivek-cdot) [reply GCP_cloud_functions/lake-to-data-warehouse/lake-to-data-warehouse.py:9]: Shoot, you're totally right. Let me get my code versions on Github and my local laptop synced.

File: GCP_cloud_functions/rsu-to-raw-ingest/rsu-to-raw-ingest.py (hunk 1 above)
Comment (payneBrandon) [reply GCP_cloud_functions/rsu-to-raw-ingest/rsu-to-raw-ingest.py:19]: If the Cloud Function failed to deploy, this code should never get called. The only additional code being protected by this is lines 21-23, which documentation on them doesn't mention any exceptions for.

Comment (dhivahari-vivek-cdot) [reply GCP_cloud_functions/rsu-to-raw-ingest/rsu-to-raw-ingest.py:19]: ...No, you're totally right there. Huh. One moment...

File: GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py (hunk 4 above)
Comment (dhivahari-vivek-cdot) [reply GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py]: Yup, forgot to sync. Fixed!

File: GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py (hunk 5 above)
Comment (dhivahari-vivek-cdot) [reply GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py]: Yup, forgot to sync. Fixed!

File: GCP_cloud_functions/lake-to-data-warehouse/lake-to-data-warehouse.py (hunk 6 above)
Comment (dhivahari-vivek-cdot) [reply GCP_cloud_functions/lake-to-data-warehouse/lake-to-data-warehouse.py]: Fixed!

File: GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py (hunk 2 above)
Comment (dhivahari-vivek-cdot) [reply GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py]: Fixed.

File: GCP_cloud_functions/rsu-to-raw-ingest/rsu-to-raw-ingest.py (hunk 1 above)
Comment (dhivahari-vivek-cdot) [reply GCP_cloud_functions/rsu-to-raw-ingest/rsu-to-raw-ingest.py:24]: Fixed!

File: GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py (hunk 3 above)
Comment (dhivahari-vivek-cdot) [reply GCP_cloud_functions/raw-to-data-lake/raw-to-data-lake.py]: Fixed!

File: GCP_cloud_functions/lake-to-data-warehouse/lake-to-data-warehouse.py (hunk 7 above)
Comment (dhivahari-vivek-cdot) [reply GCP_cloud_functions/lake-to-data-warehouse/lake-to-data-warehouse.py]: Fixed!

**Review by payneBrandon — APPROVED** [review]
Looks good!

**Pull request merged**